// ngx-drag-drop/esm2020/lib/dnd-handle.directive.mjs
import { Directive, HostBinding, HostListener } from "@angular/core";
import * as i0 from "@angular/core";
import * as i1 from "./dnd-draggable.directive";
export class DndHandleDirective {
    constructor(parent) {
        this.draggable = true;
        parent.registerDragHandle(this);
    }
    onDragEvent(event) {
        event._dndUsingHandle = true;
    }
}
DndHandleDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.3", ngImport: i0, type: DndHandleDirective, deps: [{ token: i1.DndDraggableDirective }], target: i0.ɵɵFactoryTarget.Directive });
DndHandleDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "15.0.3", type: DndHandleDirective, selector: "[dndHandle]", host: { listeners: { "dragstart": "onDragEvent($event)", "dragend": "onDragEvent($event)" }, properties: { "attr.draggable": "this.draggable" } }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.3", ngImport: i0, type: DndHandleDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: "[dndHandle]"
                }]
        }], ctorParameters: function () { return [{ type: i1.DndDraggableDirective }]; }, propDecorators: { draggable: [{
                type: HostBinding,
                args: ["attr.draggable"]
            }], onDragEvent: [{
                type: HostListener,
                args: ["dragstart", ["$event"]]
            }, {
                type: HostListener,
                args: ["dragend", ["$event"]]
            }] } });
